### PR TITLE
SIL: Lower the `self` argument of an @objc protocol class method requirement as an @objc_metatype.

### PR DESCRIPTION
**Pull request by jckarter**
Fixes rdar://problem/18921650.

### CODE REVIEW

**Comment by jckarter** [comment]
@swift-ci Please test

**Comment by jckarter** [comment]
@swift-ci Please test source compatibility

File: lib/SIL/Bridging.cpp [hunk 1]
```diff
@@ -159,7 +159,9 @@ Type TypeConverter::getLoweredCBridgedType(AbstractionPattern pattern,
 
   // Class metatypes bridge to ObjC metatypes.
   if (auto metaTy = t->getAs<MetatypeType>()) {
-    if (metaTy->getInstanceType()->getClassOrBoundGenericClass()) {
+    if (metaTy->getInstanceType()->getClassOrBoundGenericClass()
```
Comment (slavapestov) [inline lib/SIL/Bridging.cpp]: Can any other case come up here?

Comment (jckarter) [reply lib/SIL/Bridging.cpp]: Shouldn't be. Only class-constrained metatypes are considered representable in ObjC at the AST level.

**Pull request merged**